src/time-picker/TimePicker.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { props, t, skinnable } from '../utils';
+import cx from 'classnames';
+import Dropdown from '../dropdown/Dropdown';
+import FlexView from '../flex/FlexView';
+import range from 'lodash/utility/range';
+import flatten from 'lodash/array/flatten';
+
+const H24 = '24h';
+const H12 = '12h';
+const symbolRegex = /[\s,\.\|\/\\]/;
+const numberRegex = /^\d+$/;
+const separator = ':';
+
+const pad = (num) => num <= 9 ? `0${num}` : num;
+const timeFormatter24 = (hour, minute) => `${pad(hour)}:${pad(minute)}`;
+const timeFormatter12 = (hour, minute) => {
+  if (hour === 0) {
+    return `12${separator}${pad(minute)} am`;
+  } else if (hour === 12) {
+    return `12${separator}${pad(minute)} pm`;
+  } else if (hour >= 13) {
+    return `${pad(hour % 12)}${separator}${pad(minute)} pm`;
+  } else {
+    return `${pad(hour)}${separator}${pad(minute)} am`;
+  }
+};
+
+const compareTime = (minTime, maxTime) => (
+  maxTime.hours > minTime.hours || (maxTime.hours === minTime.hours && maxTime.minutes > minTime.minutes)
+);
+
+const insertColon = str => {
+  if (numberRegex.test(str) && (str.length >= 3 && str.length <= 4)) {
+    const position = str.length - 2;
+    const strWithColon = `${str.substr(0, position)}${separator}${str.substr(position)}`;
+    return strWithColon;
+  }
+  return str;
+};
+const cleanString = str => str.replace(symbolRegex, ':');
+const cleanFilter = filterStr => insertColon(cleanString(filterStr));
+const filterOptions = (options, filterStr) => (
+  options.filter( option => (
+      option.value.substr(0, filterStr.length) === cleanFilter(filterStr) ||
+      option.label.substr(0, filterStr.length) === cleanFilter(filterStr) ||
+      option.label === cleanFilter(filterStr) ||
+      option.value === cleanFilter(filterStr)
+  ))
+);
+
+const Integer = t.refinement(t.Number, n => n % 1 === 0, 'Integer');
+const Hour = t.refinement(Integer, int => int >= 0 && int <= 23, 'Hour');
+const Minute = t.refinement(Integer, int => int >= 0 && int <= 59, 'Minute');
+const TimeFormat = t.enums.of([H12, H24], 'TimeFormat');
+const Time = t.struct({
+  hours: Hour,
+  minutes: Minute
+}, 'Time');
+
+const Props = t.refinement(t.struct({
+  onChange: t.Function,
+  value: t.maybe(Time),
+  minTime: t.maybe(Time),
+  maxTime: t.maybe(Time),
+  interval: t.maybe(Integer),
+  placeholder: t.maybe(t.String),
+  timeFormat: t.maybe(TimeFormat),
+  id: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+}), ({ value, minTime, maxTime }) => (
+  compareTime(minTime, maxTime) && (!value || (compareTime(value, maxTime) && compareTime(minTime, value)))
```

Comment:
`compareTime` is not explicative as name for this function IMO
something like `lte` or `lowerThanOrEqual` would be better